src/main/host/thread_preload.c

```diff
@@ -52,6 +52,11 @@ static ThreadPreload* _threadToThreadPreload(Thread* thread) {
 
 static Thread* _threadPreloadToThread(ThreadPreload* thread) { return (Thread*)thread; }
 
+static void _threadpreload_continuePlugin(ThreadPreload* thread, const ShimEvent* event) {
+    host_unlockShimShmemLock(thread->base.host);
+    shimevent_sendEventToPlugin(thread->ipc_data, event);
+}
```

Comment:
Maybe add comments here about where this was expected to have been locked, since it can be difficult to follow the locked/unlocked state of a lock when the locking and unlocking are in different parts of the code.

Reply:
Done